src/fedora/41/amd64/Dockerfile

```diff
@@ -0,0 +1,88 @@
+FROM library/fedora:41
+
+# Install the base toolchain we need to build anything (clang, cmake, make and the like)
+# this does not include libraries that we need to compile different projects, we'd like
+# them in a different layer.
+RUN dnf --setopt=install_weak_deps=False install -y \
+        clang \
+        cmake \
+        dnf-plugins-core \
+        findutils \
+        gdb \
+        glibc-langpack-en \
+        lldb-devel \
+        llvm-devel \
+        make \
+        pigz \
+        python \
+        which \
+    && dnf clean all
+
+# Add MS package repo.
+RUN dnf5 config-manager addrepo \
```

Comment:
@omajid - I would appreciate your eyes on this.  Is this the best way to register package.microsoft.com with dnf 5?

Reply:
The [docs](https://dnf5.readthedocs.io/en/latest/changes_from_dnf4.7.html#changes-to-individual-commands) say that the replacement for `dnf config-manager --add-repo` is `dnf5 config-manager addrepo`. So this does seem to be the correct 1-to-1 port of the old dnf command.

Reply:
I see that there's a repo file already at https://packages.microsoft.com/fedora/41/prod/config.repo. Can/should we download and use that directly instead of duplicating it here?

Reply:
I missed that PMC had a config.repo file.  I will use that.  Thanks.